### DISCUSSION TITLE
How to fetch modules from private Go repositories?

### DISCUSSION THREAD

**Discussion by DCKcode** [renovatebot/renovate#10864] [Request Help]
I'm trying to get Renovate to pull dependency information from a private Golang repository. As per [the docs](https://docs.renovatebot.com/golang/#custom-registry-support-and-authentication), I've set up my `config.js` as follows:

```js
module.exports = {
    hostRules: [
        {
            matchHost: 'go.company.biz',
            token: process.env.GO_REPO_TOKEN,
        },
    ],

    golang: {
        registryUrls: [ 'go.company.biz' ],
    },
};

```

However, Renovate logs the following:
```
 WARN: Custom registries are not allowed for this datasource and will be ignored (repository=biz/test)
       "datasource": "go",
       "registryUrls": ["go.company.biz"]
```

I realize I'm asking this the day after this was merged, but I am using the latest Renovate docker image - `25.56` at this time. Is there anything I'm overlooking here?

This logline appears regardless of whether I set up authentication in `hostRules` or not, so I think it has nothing to do with that.

**Comment by nabowler** [comment]
I _think_ https://github.com/renovatebot/renovate/blob/b8e36daa87e4275caa5b66bffa6fe3f4d5d8a77e/lib/datasource/go/index.ts#L15 needs changed.

**Reply by DCKcode** [reply]
Thanks, that makes sense. Is that something that's user controllable? Or does this need a change in renovate itself?

**Reply by viceice** [reply]
I don't think that this is the problem. Renovate detects the registry url from the go dependency. I think you only need the hostrules. You can't change the registry url of a go dep, as the registry url is included in the dep name.

So we need reproduction repo. You can also check the debug logs and past them here.

**Reply by DCKcode** [reply]
This affects a private repository I'm not currently at liberty to share, unfortunately. I have debug logs enabled, but they're not showing anything.

> Renovate detects the registry url from the go dependency. I think you only need the hostrules.

When I remove the `registryUrls` configuration:

```
INFO: Unsupported go host - cannot look up versions (repository=biz/test)
       "lookupName": "go.company.biz/example"
```
That error is in Renovate [here](https://github.com/renovatebot/renovate/blob/main/lib/datasource/go/index.ts#L153). So it does look like that configuration is needed to trigger fetching from it?

**Reply by viceice** [reply]
Ok, so we made some wrong assumptions.

On update artifacts, renovate will never pass all registry urls, i stead it will ever only pass urls of the updated dependencies.

You don't want to override registry urls for public go deps, as for sure they aren't all available on your private go registry.

So we need to change the implementation. I've already a idea for it, so will try to send a fix PR tomorrow.

**Reply by DCKcode** [reply]
Ah! I'm trying to do something that is not the intended goal of this functionality?

Really appreciate the quick responses and efforts 🙂

**Reply by DCKcode** [reply]
So just to confirm I'm not missing something: Renovate currently doesn't support pulling release information from private Go repositories?

**Reply by rarkins** [reply]
Not unless they're on github.com and in the same org

**Reply by DCKcode** [reply]
Thanks for confirming. Would it help if I open a feature request for this use case 🙂 ?

**Reply by rarkins** [reply]
Sure, if it's not already captured by one of these: https://github.com/renovatebot/renovate/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Amanager%3Agomod

**Reply by DCKcode** [reply]
Thanks - I think this issue covers it: https://github.com/renovatebot/renovate/issues/7361
